---
title: "ATACseqTutorial"
author: "Ashley S Doane"
date: "November 19, 2015"
output: 
  ioslides_presentation: 
    fig_height: 6
    fig_width: 8
    highlight: zenburn
    smaller: yes
    widescreen: yes
---

```{r set-options, echo=FALSE, cache=FALSE}
options(width = 1200)
```

## ATAC-seq Tutorial

- Data set and R code downloaded here: <http://doaneas.github.io/ATACseqCSHL/>
- Open RStudio, and open the project ***ATACseqCSHL.Rproj***
- open the file ***ATAC-CSHL/annotate.R***

Install/update Bioconductor and packages

========================================================

- Bioconductor v 3.2
```{r, echo=FALSE}
source("atacseqFuncitons.R")
```


```{r, eval=FALSE}
source("atacseqFuncitons.R")
source("https://bioconductor.org/biocLite.R")

#will install biocond if not found, and update packages
#biocLite()

#if needed (biocLite will tell you)
#biocLite("BiocUpgrade") 

```

========================================================
Packages

- If library not found, install with biocLite()

```{r}

#biocLite("org.Hs.eg.db")
#biocLite("SRAdb")
library(ChIPpeakAnno)
#biocLite("org.Hs.eg.db")
#biocLite("TxDb.Hsapiens.UCSC.hg19.knownGene")
library(org.Hs.eg.db)
library(rtracklayer)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
```

========================================================


```{r}

setwd("ATAC-CSHL")

##set up to read Bed 6+4 narrowPeak file as output by MACS2.1

extraCols_narrowPeak <- c(peakName = "character", DisScore = "numeric", strand="character", foldChange = "numeric",
                          neg.logPv="numeric", neglogQval= "numeric", distPeakStart= "numeric")

#import the peaks file
atacpeaks <- import.bed("LY1.chr18.narrowPeak", extraCol= extraCols_narrowPeak)

```

========================================================

## Look at the file


```{r}
atacpeaks


```

========================================================
- what do the different coluns mean?


```{r}
mcols(atacpeaks)


```


========================================================

## Annootation of peaks

```{r}


#look at it
#what do the different columns mean?



library(TxDb.Hsapiens.UCSC.hg19.knownGene)
txdb = TxDb.Hsapiens.UCSC.hg19.knownGene
ge <- genes(txdb, columns=c("tx_name", "gene_id", "tx_type"))

anno = annotatePeakInBatch(atacpeaks, FeatureLocForDistance = "TSS",  
                                     PeakLocForDistance="middle", select = "first", 
                                     output="nearestLocation",  AnnotationData=genes(txdb))

#add gene symbols ##
anno <- addGeneIDs(anno, silence=TRUE, orgAnn="org.Hs.eg.db", feature_id_type="entrez_id", IDs2Add=c("symbol"))


annoclose <- anno[abs(anno$distancetoFeature) < 2000]

unique(na.omit(as.vector(annoclose$symbol)))


pie(table(anno$insideFeature))

### chromosomal regions ###

aCR<-assignChromosomeRegion(anno, nucleotideLevel=FALSE, 
                            proximal.promoter.cutoff=2000L,
                            immediate.downstream.cutoff=2000L,
                            precedence=c("Promoters", "immediateDownstream", 
                                         "fiveUTRs", "threeUTRs", 
                                         "Exons", "Introns"), TxDb=txdb)

#plot the percentages

barplot(aCR$percentage)
```

========================================================

## Plot the % of total peaks in each region

```{r, echo=FALSE}
barplot(aCR$percentage)
```


## Install/update Bioconductor and packages
- Bioconducor

```{r}
source("atacseqFuncitons.R")
source("https://bioconductor.org/biocLite.R")
#biocLite()
#if needed:
#biocLite("BiocUpgrade") 
```



```{r}

#biocLite("org.Hs.eg.db")
#biocLite("SRAdb")
library(ChIPpeakAnno)
#biocLite("org.Hs.eg.db")
#biocLite("TxDb.Hsapiens.UCSC.hg19.knownGene")
library(org.Hs.eg.db)
library(rtracklayer)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
```

```{r}
setwd("ATAC-CSHL")

##set up to read Bed 6+4 narrowPeak file as output by MACS2.1

extraCols_narrowPeak <- c(peakName = "character", DisScore = "numeric", strand="character", foldChange = "numeric",
                          neg.logPv="numeric", neglogQval= "numeric", distPeakStart= "numeric")

#import the peaks file
atacpeaks <- import.bed("LY1.chr18.narrowPeak", extraCol= extraCols_narrowPeak)

```

## Look at the file
- what do the different coluns mean?

```{r}
atacpeaks


```


## Annootation of peaks

```{r}


#look at it
#what do the different columns mean?



library(TxDb.Hsapiens.UCSC.hg19.knownGene)
txdb = TxDb.Hsapiens.UCSC.hg19.knownGene
ge <- genes(txdb, columns=c("tx_name", "gene_id", "tx_type"))

anno = annotatePeakInBatch(atacpeaks, FeatureLocForDistance = "TSS",  
                                     PeakLocForDistance="middle", select = "first", 
                                     output="nearestLocation",  AnnotationData=genes(txdb))

#add gene symbols ##
anno <- addGeneIDs(anno, silence=TRUE, orgAnn="org.Hs.eg.db", feature_id_type="entrez_id", IDs2Add=c("symbol"))


annoclose <- anno[abs(anno$distancetoFeature) < 2000]

unique(na.omit(as.vector(annoclose$symbol)))


pie(table(anno$insideFeature))

### chromosomal regions ###

aCR<-assignChromosomeRegion(anno, nucleotideLevel=FALSE, 
                            proximal.promoter.cutoff=2000L,
                            immediate.downstream.cutoff=2000L,
                            precedence=c("Promoters", "immediateDownstream", 
                                         "fiveUTRs", "threeUTRs", 
                                         "Exons", "Introns"), TxDb=txdb)

#plot the percentages

barplot(aCR$percentage)
```

## Plot the % of total peaks in each region

```{r, echo=FALSE}
barplot(aCR$percentage)
```

